### PR TITLE
Launch product feedback survey

### LINKED ISSUE #2648: [IAFeedback] Launch beta product survey from Product Banner
Product Survey should be launched modally from the product banner

Screen Shot 2020-08-12 at 4 32 23 PM

Tasks:
- [x] Get production Survey URL
- [ ] Present product feature survey from the "Give Feedback" CTA
- [ ] Dismiss(And don't show again) the banner after tapping on any CTA

### PR DESCRIPTION
**Pull request by Ecarrion**
closes #2648

# Why
In order to complete the product feedback survey flow this PR:
- Adds product feedback survey URL
- Updates products banner to include "Give Feedback" and "Dismiss" action button
- Launch the product survey when tapping on the "Give Feedback" CTA
- Dismiss product banner when tapping on the "Dismiss" CTA
- Update `ProductsViewController` to query the product feedback visibility to know if the banner has to be shown or hidden.

# Screenshots

Dismiss | Give Feedback
---- | ----
dismiss | give-feedback

# Testing Steps

## Give Feedback
- Delete the app to reset any saved settings
- Open the app and login into a store
- Navigate to products
- Tap on the collapsed products banner
- Tap on "Give Feedback" and see that the product feedback is presented
- Complete the feedback and see that the product banner is hidden
- Relaunch the app and see that the product banner is not shown

## Dismiss
- Delete the app to reset any saved settings
- Open the app and login into a store
- Navigate to products
- Tap on the collapsed products banner
- Tap on "Dismiss" and see that the banner is dismissed
- Relaunch the app and see that the product banner is not shown

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Products/ProductsTopBannerFactory.swift [hunk 1]
```diff
@@ -6,19 +6,29 @@ struct ProductsTopBannerFactory {
     /// - Parameters:
     ///   - isExpanded: whether the top banner is expanded by default.
     ///   - expandedStateChangeHandler: called when the top banner expanded state changes.
+    ///   - onGiveFeedbackButtonPressed: called the give feedbak button is pressed.
+    ///   - onDismisskButtonPressed: called the dismiss button is pressed
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Products/ProductsTopBannerFactory.swift]:
```suggestion
    ///   - onDismissButtonPressed: called the dismiss button is pressed
```

File: WooCommerce/Classes/ViewRelated/Products/ProductsTopBannerFactory.swift [hunk 2]
```diff
@@ -6,19 +6,29 @@ struct ProductsTopBannerFactory {
...
     ///   - onCompletion: called when the view controller is created and ready for display.
     static func topBanner(isExpanded: Bool,
                           stores: StoresManager = ServiceLocator.stores,
+                          isInAppFeedbackFeatureEnabled: Bool,
                           expandedStateChangeHandler: @escaping () -> Void,
+                          onGiveFeedbackButtonPressed: @escaping () -> Void,
+                          onDismisskButtonPressed: @escaping () -> Void,
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Products/ProductsTopBannerFactory.swift]:
```suggestion
                          onDismissButtonPressed: @escaping () -> Void,
```

**Review by jaclync — APPROVED** [review]
LGTM! :shipit: I also requested G's review since it has the new banner icon

Code: ✅ 
Testing: ✅ (there's a little gap above the Filter/Sort banner, pretty minor though)

**Comment by jaclync** [comment]
> I've fixed the little gap problem too.

@Ecarrion the fix looks great in the GIF! just wondering, did you make a commit for this fix? 🤔 

regarding design review, I wasn't sure if we plan to launch the product feedback survey in release 4.9/5.0 - if you've planned to request a design review separately, feel free to remove the design review from the PR 🙂

**Review by Garance91540 — COMMENTED** [review]
This is awesome, thank you for working on this @Ecarrion ! Do you have some dark mode screenshots too so I can take a quick look?

**Comment by Ecarrion** [comment]
@jaclync Yes, the fix was made here https://github.com/woocommerce/woocommerce-ios/pull/2686/commits/6c073ad1bf50e523dd9ebaa4b37b1a34e9b6a784#diff-3ffb83b7414d96dc7a388662e8304143R406 It did slip with the wrong message tho 🤦 

@Garance91540 Here is a dark mode screenshot for you to look at
Screen Shot 2020-08-24 at 7 25 36 AM

**Review by Garance91540 — APPROVED** [review]
Looking good! Can we update the speaker icon on dark mode so it's Purple 30?

**Comment by Ecarrion** [comment]
@Garance91540 Sure!
dark

**Pull request merged**